libraries/abstractions/common_io/test/test_scripts/uart/test_iot_uart_test.py

```diff
@@ -0,0 +1,259 @@
+        self._serial.write(cmd.encode("utf-8"))
+        self._serial.write("\r\n".encode("utf-8"))
+
+        dut_res =  str(self._serial.read_until(terminator=serial.to_bytes([ord(c) for c in 'Ignored \n\r'])))
+        dut_result = dut_res_pattern.search(dut_res)
+
+        print(dut_res)
+        t_shell.join()
+
+        with open(self.rpi_output_file, "r") as f:
+            result_matches = pi_res_pattern.findall(str(f.read()))
+        s.send(b'E')
+        s.close()
+        self.clean()
+
+        return "Pass" if len(result_matches) >= 2 and dut_result else "Fail"
+
+    def test_IotUartWriteAsync(self):
+        """
+        Test body of uart write async test. Return pass if expected message is received
+        :return: 'Pass' or 'Fail'
+        """
+        pi_result = None
+        dut_result = None
+        dut_res_pattern = re.compile(r"AFQP_AssistedIotUARTWriteAsync\W\sPASS")
+        pi_res_pattern = re.compile(r"\b(?:\\xaa){198}\b")
+
+        t_shell = threading.Thread(target=self.run_shell_script,
+                                   args=(" ".join([self.shell_script, self._ip, self._login, self._pwd, '3']),))
+        t_shell.start()
+
+        # IMPORTANT: without this, the socket won't connect on mac; reason not known.
+        sleep(1)
+
+        # Create a tcp type socket with AF_INET address family.
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        time_out = 10
+
+        # Wait until connection with the process on rpi is established.
+        while s.connect_ex((self._ip, self.port)) != 0 and time_out > 0:
+            time_out -= 1
+            sleep(1)
+        if time_out == 0:
+            print("Socket connection cannot be established")
+            s.close()
+            return "Fail"
+
+        print("connnected successfully")
+
+        self._serial.reset_input_buffer()
+
+        cmd = "iot_tests test 1 3"
+        self._serial.write("\r\n".encode("utf-8"))
+        self._serial.write(cmd.encode("utf-8"))
+        self._serial.write("\r\n".encode("utf-8"))
+
+        dut_res =  str(self._serial.read_until(terminator=serial.to_bytes([ord(c) for c in 'Ignored \n\r'])))
+        dut_result = dut_res_pattern.search(dut_res)
+        print(dut_res)
+        t_shell.join()
+
+        with open(self.rpi_output_file, "r") as f:
+            rpi_output = str(f.read());
+            print(rpi_output)
+            pi_result = pi_res_pattern.search(rpi_output)
+
+        s.send(b'E')
+        s.close()
+        self.clean()
+
+        return "Pass" if pi_result and dut_result else "Fail"
+
+    def clean(self):
+        os.remove("./uart_res.txt")
+        self.run_shell_script(
+            " ".join([self.shell_script, self._ip, self._login, self._pwd, "-c"])
+        )
+
+
+# unit test
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('-i', '--ip', nargs=1, default=['10.53.30.75'], help='ip address of rpi')
```

Comment:
Could you take away the content in the default?

Reply:
Oh, sorry. Yes, I will remove